### PR TITLE
Use local clusterctl overrides for determining latest version for provider

### LINKED ISSUE #599: CAPIProvider should signal when new update is available
We can use the clusterctl/capi operator code to determine if a new update is available and reflect this in the status. Airgapped scenarios should also be supported.

### LINKED ISSUE #691: Providers version not being updated with turtles upgrade
### What steps did you take and what happened?

**Steps:**
- Install turtles v0.10.0 and enable docker, aws, gcp infra providers using capi-ui extension
- Following versions are installed: docker: v1.4.6, aws: v2.3.5, gcp: v1.6.0 (auto-installed with turtles - core: v1.4.6, rke2: v0.4.1, addon: v0.3.1)

image

- Upgrade turtles to v0.11.0 and check provider versions, core and rke2 versions are upgraded along with turtles version upgrade and not other providers (core: v1.7.3, rke2: v0.5.0)

image

### What did you expect to happen?

Installed CAPI Providers should upgrade if newer version is available, along with turtles upgrade

### How to reproduce it?

Steps mentioned above.

### Rancher Turtles version

v0.11.0

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #599
Fixes #691 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: internal/sync/provider_sync.go [hunk 1]
```diff
@@ -152,31 +160,48 @@ func (s *ProviderSync) rolloutInfrastructure() {
...
+			turtlesv1.CheckLatestUpdateAvailableReason,
+			clusterv1.ConditionSeverityInfo,
+			"Provider version update available. Current latest is %s", providerVersion,
+		)
+	case !latest:
+		lastCheck := conditions.Get(s.Source, turtlesv1.CheckLatestVersionTime)
+		updatedMessage := fmt.Sprintf("Updated to latest %s version", providerVersion)
+
+		if lastCheck == nil || lastCheck.Message != updatedMessage {
+			log.Info(fmt.Sprintf("Version %s is beyound current latest, updated to %s", cmp.Or(s.Source.Spec.Version, "latest"), providerVersion))
```
Comment (furkatgofurov7) [inline internal/sync/provider_sync.go:195]:
```suggestion
			log.Info(fmt.Sprintf("Version %s is beyond current latest, updated to %s", cmp.Or(s.Source.Spec.Version, "latest"), providerVersion))
```

File: api/v1alpha1/conditions_consts.go [hunk 2]
```diff
@@ -34,3 +34,11 @@ const (
 	// CheckLatestVersionTime is set as a timestamp info of the last timestamp of the latest version being up-to-date for the CAPIProvider.
 	CheckLatestVersionTime = "CheckLatestVersionTime"
 )
+
+const (
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply api/v1alpha1/conditions_consts.go:38]: It is easy to mix the condition and the reason, so I separated these out, similar to upstream https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/condition_consts.go

[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply api/v1alpha1/conditions_consts.go:38]: Makes sense then

**Review by furkatgofurov7 — DISMISSED** [review]

**Review by salasberryfin — APPROVED** [review]
Thanks, LGTM.

**Review by salasberryfin — APPROVED** [review]

File: internal/sync/provider_sync_test.go [hunk 3]
```diff
@@ -104,16 +128,150 @@ var _ = Describe("Provider sync", func() {
...
+			g.Expect(err).To(Succeed())
+			g.Expect(conditions.IsTrue(expected, turtlesv1.LastAppliedConfigurationTime)).To(BeTrue())
+			g.Expect(conditions.IsFalse(expected, turtlesv1.CheckLatestVersionTime)).To(BeTrue())
+		}).Should(Succeed())
+
+		Eventually(Object(customInfrastructure)).Should(
+			HaveField("Spec.ProviderSpec", Equal(expected.Spec.ProviderSpec)))
+		Consistently(Object(customInfrastructure)).Should(
+			HaveField("Spec.ProviderSpec", Equal(expected.Spec.ProviderSpec)))
+	})
```
Comment (furkatgofurov7) [inline internal/sync/provider_sync_test.go:272]: this looks bad copy&paste of above test and can be dropped

**Review by furkatgofurov7 — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
@salasberryfin can you please re-approve?

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**